sysconfig/cloudinit.go

```diff
@@ -263,6 +263,68 @@ func cloudDatasourcesInUse(configFile string) (*cloudDatasourcesInUseResult, err
 	return res, nil
 }
 
+// cloudDatasourcesInUseForDir considers all files in a directory as individual
+// cloud-init config files, and analyzes all datasources in use for each file
+// and returns their union. It does not distinguish between mentioned,
+// explicitly allowed, or explicitly disallowed, but it does follow cloud-init's
+// logic for determining the overwriting of properties such that if a lexcially
+// earlier file sets datasource_list: [], then a later file sets
+// datasource_list: [foo], then foo is used instead and the explicit disallowing
+// is ignored/overwritten, but if no other file sets datasource_list, then it is
+// treated as if the config explicitly disallows no datasources.
+func cloudDatasourcesInUseForDir(dir string) (*cloudDatasourcesInUseResult, error) {
+
+	files, err := filepath.Glob(filepath.Join(dir, "*"))
```

Comment:
does cloud-init or should we filter based on extension?

Reply:
hmm, well we do have an existing restriction of only installing files that end in `.cfg` but I'm not sure what cloud-init does/feels about this, I will have to look

Reply:
it took me a while to get back to this, but indeed ds-identify at least restricts itself to considering only .cfg files in the .d directory, so we should also restrict ourselves to .cfg too. I will push a patch for this